src/client/desktop_gui/lib/shell_terminal.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_pty/flutter_pty.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:xterm/xterm.dart';
+
+import 'settings.dart';
+
+const terminalFontSizeKey = 'terminalFontSize';
+
+class ShellTerminal extends ConsumerStatefulWidget {
+  final String instance;
+
+  const ShellTerminal({super.key, required this.instance});
+
+  @override
+  ConsumerState<ShellTerminal> createState() => _ShellTerminalState();
+}
+
+class _ShellTerminalState extends ConsumerState<ShellTerminal> {
+  final terminal = Terminal(maxLines: 10000);
```

Comment:
I don't know if it pre-allocates, but I'll look into int. The Workshops project also uses 10k for their Terminal, so that's why I used that value. But I'll play around with the value and see the effects it has.

Reply:
OK sounds good, thanks! In any case, no need to spend a lot of time looking into it in depth. We can always reduce if we find we need to, with no significant loss to users.

Reply:
> I can confirm that it pre-allocates :/

Oh OK, I hadn't seen your reply. So, depending on what you observe, perhaps 50K lines? That should still allow creating ~10 terminals before hitting the 100M bar. As long as that garbage collector isn't too lazy...